apps/fz_http/lib/fz_http_web/plug/path_prefix.ex

```diff
@@ -0,0 +1,32 @@
+defmodule FzHttpWeb.Plug.PathPrefix do
+  @moduledoc """
+  This Plug removes prefix from Plug.Conn path fields which allows to run Firezone
+  under non root directory without recompiling it.
+  """
+  @behaviour Plug
+
+  def init(opts), do: opts
+
+  def call(%Plug.Conn{path_info: path_info, request_path: request_path} = conn, _opts) do
+    path_prefix = fetch_path_prefix!()
+    request_path_info = String.split(request_path, "/")
+    {path_info, request_path_info} = trim_prefix(path_info, request_path_info, path_prefix)
+    %{conn | path_info: path_info, request_path: Enum.join(request_path_info, "/")}
+  end
+
+  defp fetch_path_prefix!() do
+    case Application.fetch_env!(:fz_http, :path_prefix) do
+      prefix when is_binary(prefix) -> String.trim(prefix, "/")
+      _other -> nil
+    end
+  end
+
+  defp trim_prefix(path_info, request_path_info, nil),
+    do: {path_info, request_path_info}
+
+  defp trim_prefix([prefix | path_info], ["", prefix | request_path_info], prefix),
+    do: {path_info, [""] ++ request_path_info}
+
+  defp trim_prefix(path_info, request_path_info, _prefix),
```

Comment:
ATM this is very permissive, so if the configuration is set for `/vpn` for example Firezone will keep working both at root `/` and at `/vpn`, should I be more restrictive and return 404 if somebody tries to hit the root?

Reply:
I think a middle-ground option would be returning 301 Moved Permanently to the EXRTERNAL_URL. But 404 should be good too.